lib/collab_migration.rb

```diff
@@ -34,7 +35,7 @@ def github_organization
   end
 
   def github_user
-    @github_user ||= GitHubUser.new(@repo_access.user.github_client, @repo_access.user.uid)
+    @github_user ||= GitHubUser.new(@user.github_client, @user.uid, classroom_resource: @user)
```

Comment:
This is probably something completely obvious I'm overlooking and more out curiosity, but why are we passing in a `classroom_resource` reference to the classroom user?  I'd imagine its because that's where we're cacheing the fields, but the only changes I see in reference to it is deleting it within the [initializer for GitHubModel](https://github.com/education/classroom/pull/1964/files#diff-6a9532badda0e079e68cf8a3cbdf8766R37)

Reply:
That `delete` serves two purposes. One, it removes the key from the hash, but `Hash#delete` also returns the value deleted. So `resource = options.delete(:classroom_resource)` removes the key from the hash and assigns the value to `resource`. We then use resource when building the cache here: https://github.com/education/classroom/pull/1964/files#diff-6a9532badda0e079e68cf8a3cbdf8766R69